### DISCUSSION TITLE
Spore Meeting Agenda @ 2024-01-22

### DISCUSSION THREAD

**Discussion by ashuralyk** [sporeprotocol/spore-scrum#17] [Show and tell]
# Goal
[[previous]](https://github.com/sporeprotocol/spore-scrum/discussions/14)

1. mainnet launch, see checklist https://github.com/sporeprotocol/spore-scrum/issues/6#issuecomment-1901773073
2. help `Philosopher Stone` successfully run for a couple of days
3. fix bugs in `spore-demo` (low priority)

talks: 
* I plan to use the normal deployment method to upload Spore contracts which is not a ZERO lock and can be consumed potentially, is that ok for now? @Flouse @RongxinZhang 

# Previous Work

1. co-build integration within `spore-contract` and `spore-sdk` has been finished, see CI workflow here https://github.com/sporeprotocol/spore-contract/pull/28

2. progress on `spore-contract`
    - [x] fix a compatibility bug https://github.com/sporeprotocol/spore-contract/pull/50
    - [x] complete test cases https://github.com/sporeprotocol/spore-contract/pull/44
    - [x] manage docs https://github.com/sporeprotocol/spore-contract/pull/48
    - [x] release tags `v0.1.0` and `v0.2.0`

3. progress on `spore-sdk`
    - [x] integrate co-build https://github.com/sporeprotocol/spore-sdk/pull/68
    - [x] remove `minPayment` argument from transfer ClusterProxy https://github.com/sporeprotocol/spore-sdk/pull/69
    - [x] dump version to `0.2.0-beta.1` https://github.com/sporeprotocol/spore-sdk/pull/74

5. fix bugs in `spore-graphql` after the breaking change of `spore-sdk`
    - [x] https://github.com/sporeprotocol/spore-graphql/pull/3
    - [x] https://github.com/sporeprotocol/spore-graphql/pull/2

6. update dependencies in `spore-demo` and give some manual tests
    - [x] collect issues from testing https://github.com/sporeprotocol/spore-demo/issues/54#issuecomment-1898207033
    - [ ] fix bugs

# Next Plan
1. carry out a more proper version management on `spore-contract` to eliminate further chaos
2. deploy `spore-contract` on mainnet
3. help Philosopher Stone complete its test

--------
cc @RongxinZhang @Flouse @Dawn-githup @ShookLyngs

**Comment by ShookLyngs** [comment]
Done:
- spore-sdk
  - https://github.com/sporeprotocol/spore-sdk/pull/69
  - https://github.com/sporeprotocol/spore-sdk/pull/71
  - https://github.com/sporeprotocol/spore-sdk/pull/73
  - https://github.com/sporeprotocol/spore-sdk/pull/74
  - https://github.com/sporeprotocol/spore-sdk/pull/75
  - https://github.com/sporeprotocol/spore-sdk/pull/76
- spore-demo
  - https://github.com/sporeprotocol/spore-demo/pull/57
  - https://github.com/sporeprotocol/spore-demo/pull/59
  - https://github.com/sporeprotocol/spore-demo/pull/61
  - https://github.com/sporeprotocol/spore-demo/pull/63
- spore-graphql
  - https://github.com/sporeprotocol/spore-graphql/pull/4

Next:
- Test spore-sdk#76 and publish a new version
- Review spore-graphql code, if nothing more, publish a new version

**Comment by Flouse** [comment]
> * I plan to use the normal deployment method to upload Spore contracts which is not a ZERO lock and can be consumed potentially, is that ok for now? @Flouse @RongxinZhang

The `capsule build --release` cmd of spore-contract is a reproducible build solution.
So, technically, anyone can deploy a special version of code with the same code_hash on CKB chain.

This also means you can make a deployment with ZERO lock at any time when you think it's the right time.

## Related Suggestions
1. ### Clearer disclosure of deployment information 
   Update the deployment info in spore-contract repo's README or RFC, including:
   - network (CKB testnet or mainnet)
   - spore-contract's version tag or commit
   - contract_name
   - code_hash
   - hash_type
   - tx_hash
   - index
   - dep_type

2. add docs in README: explain that reproducible build is supported to verify the deploy scripts/contracts

## References
- https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0025-simple-udt/0025-simple-udt.md#notes

**Reply by ashuralyk** [reply]
> Update the deployment info in spore-contract repo's README or RFC

after things go well, I'll update these information in README.md and `docs/VERSIONS.md`

**Comment by Dawn-githup** [comment]
Done:
  
  1. Conduct usability testing on spore-demo based on the latest spore-contract.https://github.com/sporeprotocol/spore-scrum/issues/8#issuecomment-1888480013
  2. Add spore_v1 contract https://github.com/sporeprotocol/spore-devenv/pull/4
  3. spore demo/spore graphql problem repair verification https://github.com/sporeprotocol/spore-demo/pull/61#issue-2089663295
  
Next:

  1. spore-demo problem fix verification https://github.com/sporeprotocol/spore-demo/issues/created_by/Dawn-githup
  2. Verification based on new version of spore-graphql https://github.com/sporeprotocol/spore-demo/issues/54#issuecomment-1898207033

**Comment by anipenguin** [comment]
Done:
- Linked to X@sporeprotocol on the official website [spore.pro](http://spore.pro/). https://github.com/sporeprotocol/spore-site/pull/24

Next:
- Update the tutorial to reflect the changes with the SDK.
- Prepare relevant tweets and announcements for when the launch date is confirmed.

**Reply by ashuralyk** [reply]
> Update the tutorial to reflect the changes with the SDK

how do you know these changes in SDK?

**Reply by anipenguin** [reply]
it would be based on the updates to https://github.com/sporeprotocol/spore-blog-tutorial with help from the engineer

**Comment by ashuralyk** [comment]
# Progress of Scrum

1. mainnet deployment complete
2. `spore-sdk` adapted mainnet deployment
3. Philosopher Stone is about to switch to mainnet network
4. team breaking changed
